src/Mailgun/Model/Suppression/Unsubscribe/Unsubscribe.php

```diff
@@ -74,14 +144,24 @@ public function getTag()
     }
 
     /**
-     * @param string $tag
+     * Set the tag 
+     * 
+     * Note: * means all unsubscribe address from domain
+     * 
+     * If NULL the value means * (all for a domain)
+     * 
+     * @param string $tag The tag
      */
-    private function setTag($tag)
+    private function setTag($tag = NULL)
     {
         $this->tag = $tag;
+        
+        return $this;
```

Comment:
Do we need a fluent api for these?

Reply:
I think that doesn't hurt, in fact could be good add in other models too. It is the symfony way in setters as far I understand and a lot people expect from moderm apis the chanining of setters ;)

Reply:
I don't think we need to return `this` here, but up to you

Reply:
@DavidGarciaCat my idea (if the PR is accepted) is make new PR iterating over all the models and classes and add the chaining feature to every set method

Reply:
@shakaran although I know what you mean, I still don't see the need to add `return $this;` everywhere, specially when we're dealing with `private` methods.

Also, what happens if - a human mistake - we forget a single `return $this;` and then the SDK throws an error due we can't concatenate the next call?

Decouple matters, and may help to see what you're doing with each object - don't see the problem to use:
```php
$myAwesomeModal->setFirstValue('...');
$myAwesomeModal->setSecondValue('...');

$anotherResource->setAnotherValue('...');
```